### ISSUE TITLE
cross-validated train

### ISSUE THREAD

**Issue by sky186** [cgnorthcutt/cleanlab#52]
@cgnorthcutt 
Hello :
Thank you approach^^,  but I have a question for you.
your approach step is 4 cross-validated ,  every train  num_classes i s same ,  last to concat 4 softmax.npy,
but for any training set, if  use all data  to train ,get  last softmax.npy   is  ok?   or will have problem ?  or what kind of data does your method work well or not？

**Comment by cgnorthcutt** [comment]
hi @sky186 -- the softmax predicted probabilities must be computed out-of-sample. You cannot just train on the train set. Why? Because the model can give prob(argmax label | data) = 1 and prob(not argmax label | data) = 0 if you use the predicted probabilities on the train set -- that's literally what your loss function is making happen (if you use cross-entropy). So using the probabilities on your train set necessarily provides bad probabilities that tend toward 0 and 1 instead of tending toward the true predicted probability (which encapsulates the noise -- and the noise is what we want to model).

**Comment by sky186** [comment]
@cgnorthcutt 
谢谢您的回复
我的理解过程
1、对于任意数据，每个类别的图像分成4份， 得到4份 类别一样的 数据集（
2、三份训练 一份测试， 最后得到四个测试结果拼接成 一个矩阵，

这里有个问题，
1、数据如果很不平衡， 例如有不少只有1到2张的（低于4张），是选择全部过滤掉低于4张的类别图片， 还是随机分配图片到4个交叉验证数据中（维持类别一样） 
2、得到的概率0 1 的概率还是有但是会大量减少
3、训练的模型的精度，对于最后的过滤影响，是不是很大？
4、对于未知的数据有在训练类别的也有不在训练类别中，cleanlab 是不是 不可用
